code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +122,296 @@
+							if(L.Adjacent(src) && busy != SMASHING_LIGHT)
+								L.attack_animal(src)*/
+
+					for(var/obj/machinery/door/D in can_see)//Else, let's open some doors and see what's around
+						if(can_open_door(D) && !busy)
+							Goto(D, speed)
+							busy = MOVING_TO_TARGET
+							give_up(D)
+
+							if(D.Adjacent(src) && busy != OPENING_DOOR)
+								busy = OPENING_DOOR
+								force_door(D)
+
+		else
+			busy = 0
+			stop_automated_movement = 0
+
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/give_up(var/C)
+	spawn(100)
+		if(busy == MOVING_TO_TARGET)
+			if(target == C && get_dist(src,target) > 1)
+				target = null
+			busy = 0
+			stop_automated_movement = 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/can_open_door(var/obj/machinery/door/D)
+	if(istype(D,/obj/machinery/door/poddoor) || istype(D, /obj/machinery/door/airlock/multi_tile/glass) || istype(D, /obj/machinery/door/window))
+		return 0
+
+	// Don't fuck with doors that are doing something
+	if(D.operating>0)
+		return 0
+
+	// Don't open opened doors.
+	if(!D.density)
+		return 0
+
+	// Can't open bolted/welded doors
+	if(istype(D,/obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A=D
+		if(A.locked || A.welded || A.jammed)
+			if(break_doors == 2)
+				return 1
+			else
+				return 0
+
+	return 1
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/force_door(var/obj/machinery/door/D)
+	stop_automated_movement = 1
+	D.visible_message("<span class='warning'>\The [D]'s motors whine as something attempts to brute force their way through it!</span>")
+	playsound(get_turf(D), 'sound/effects/grillehit.ogg', 50, 1)
+	D.shake(1, 8)
+	stop_automated_movement = 1
+	spawn(100) //Roughly 10 seconds for people on the other side of the door to run the heck away
+		if(can_open_door(D))//Let's see if nobody quickly bolted it
+			if(istype (src, /mob/living/simple_animal/hostile/necro/zombie/crimson)) //Guaranteed
+				D.visible_message("<span class='warning'>\The [D] breaks open under the pressure</span>")
+				if(istype(D, /obj/machinery/door/airlock/))
+					var/obj/machinery/door/airlock/A = D
+					A.locked = 0
+					A.welded = 0
+					A.jammed = 0
+				D.open(1)
+			else
+				if(prob(15))
+					D.visible_message("<span class='warning'>\The [D] creaks open under force, steadily</span>")
+					D.open(1)
+		busy = 0
+		stop_automated_movement = 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/check_edibility(var/mob/living/carbon/human/target)
+	if(isjusthuman(target)) //Humans are always edible
+		return 1
+	if(target.health > -400) //So they're not caught eating the same dumb bird all day
+		return 1
+
+	return 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/eat(var/mob/living/carbon/human/target)
+	//Deal a random amount of brute damage to the corpse in question, heal the zombie by the damage dealt halved
+	visible_message("<span class='notice'>\the [src] starts to take a bite out of \the [target].</span>")
+	spawn(50)
```

Comment:
if(do_after(src, target, 50, needhand = FALSE))

You should really avoid a bare spawn() if you're interacting with anything other than src, because you will miss out a LOT of sanity. I can think of three things right off the bat that you've not checked that do_after will check for you.